src/OpenTelemetry.AutoInstrumentation.Native/CMakeLists.txt

```diff
@@ -96,21 +94,29 @@ SET(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${OUTPUT_BIN_DIR})
 # Dependencies
 # ******************************************************
 
-# Add custom commands to setup dependencies via git
+include(FetchContent)
 
-add_custom_command(
-    OUTPUT ${OUTPUT_DEPS_DIR}/re2/obj/libre2.a
-    COMMAND git clone -c advice.detachedHead=false --quiet --depth 1 --branch 2022-12-01 https://github.com/google/re2.git && cd re2 && env ARFLAGS=\"-r -s -c\" CXXFLAGS=\"-std=c++11 -O3 -g -fPIC\" make
-    WORKING_DIRECTORY ${OUTPUT_DEPS_DIR}
+# Fetch re2 library
+FetchContent_Declare(
+    re2
+    GIT_REPOSITORY https://github.com/google/re2.git
+    GIT_TAG 2023-06-02
 )
+FetchContent_GetProperties(re2)
+if(NOT re2_POPULATED)
+    FetchContent_Populate(re2)
+endif()
 
-add_custom_command(
-    OUTPUT ${OUTPUT_DEPS_DIR}/fmt/libfmt.a
-    COMMAND git clone -c advice.detachedHead=false --quiet --depth 1 --branch 9.1.0 https://github.com/fmtlib/fmt.git && cd fmt && cmake -DCMAKE_POSITION_INDEPENDENT_CODE=TRUE -DFMT_TEST=0 -DFMT_DOC=0 . && make
-    WORKING_DIRECTORY ${OUTPUT_DEPS_DIR}
+# Fetch fmt library
+FetchContent_Declare(
+    fmt
+    GIT_REPOSITORY https://github.com/fmtlib/fmt.git
+    GIT_TAG 10.0.0
```

Comment:
Bumping this package requires also additional changes. It was the reason I suggested to do this separately.
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/2638#issuecomment-1586967593

Reply:
> Bumping this package requires also additional changes. It was the reason I suggested to do this separately. [#2638 (comment)](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/2638#issuecomment-1586967593)
I will push seperate prs to bump the version and test the pr this week.